pkg/controller/dnszone/awsactuator.go

```diff
@@ -472,6 +472,110 @@ func (a *AWSActuator) Exists() (bool, error) {
 	return a.zoneID != nil, nil
 }
 
+func (a *AWSActuator) setInsufficientCredentialsConditionToFalse() bool {
+	accessDeniedConds, accessDeniedCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.InsufficientCredentialsCondition,
+		corev1.ConditionFalse,
+		accessDeniedReason,
+		"credentials are valid",
+		controllerutils.UpdateConditionNever,
+	)
+	if accessDeniedCondsChanged {
+		a.dnsZone.Status.Conditions = accessDeniedConds
+	}
+
+	return accessDeniedCondsChanged
+}
+
+func (a *AWSActuator) setInsufficientCredentialsConditionToTrue(message string) bool {
+	accessDeniedConds, accessDeniedCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.InsufficientCredentialsCondition,
+		corev1.ConditionTrue,
+		accessDeniedReason,
+		message,
+		controllerutils.UpdateConditionIfReasonOrMessageChange,
+	)
+
+	if accessDeniedCondsChanged {
+		// Conditions have changed. Update them in the object.
+		a.dnsZone.Status.Conditions = accessDeniedConds
+	}
+	return accessDeniedCondsChanged
+}
+
+func (a *AWSActuator) setAuthenticationFailureConditionToFalse() bool {
+	authenticationFailureConds, authenticationFailureCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.AuthenticationFailureCondition,
+		corev1.ConditionFalse,
+		authenticationFailureReason,
+		"credentials authenticated",
+		controllerutils.UpdateConditionNever,
+	)
+	if authenticationFailureCondsChanged {
+		a.dnsZone.Status.Conditions = authenticationFailureConds
+	}
+
+	return authenticationFailureCondsChanged
+}
+
+func (a *AWSActuator) setAuthenticationFailureConditionToTrue(message string) bool {
+	var authenticationFailureConds []hivev1.DNSZoneCondition
+	authenticationFailureConds, authenticationFailureCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.AuthenticationFailureCondition,
+		corev1.ConditionTrue,
+		authenticationFailureReason,
+		message,
+		controllerutils.UpdateConditionIfReasonOrMessageChange,
+	)
+
+	if authenticationFailureCondsChanged {
+		// Conditions have changed. Update them in the object.
+		a.dnsZone.Status.Conditions = authenticationFailureConds
+	}
+
+	return authenticationFailureCondsChanged
+}
+
+// SetConditionsForError sets conditions on the dnszone given a specific error. Returns true if conditions changed.
+func (a *AWSActuator) SetConditionsForError(err error) bool {
+	if err == nil {
```

Comment:
This check is not necessary. It will be covered by the check for whether the error is a `awserr.Error`.